### PR TITLE
Enable AutoTarget to aquire targets under fog if GPS is present

### LINKED ISSUE #9953: GPS - Fog of War - Unit Targeting
Long range units, such as artillery and V2s do not auto-target enemies within range when GPS ability is up, they will only fire if another unit has line of sight.  AI skirmish opponents seem to have no trouble targeting deep within the players base when they should not have line of sight. This was a change introduced with the playtest 20151031. You did not need active line of sight (cleared fog of war), just GPS was enough for artillery to fire on enemy units in previous releases.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/28116339-gps-fog-of-war-unit-targeting?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
Fixes #9953.

### CODE REVIEW

**Comment by GraionDilach** [comment]
:+1:

File: OpenRA.Mods.Common/Traits/AutoTarget.cs [hunk 1]
```diff
@@ -190,7 +190,7 @@ Actor ChooseTarget(Actor self, WDist range, bool allowMove)
 								&& !target.IsInRange(self.CenterPosition, arm.Weapon.MinRange))), a);
 				})
 
-				.Where(kv => kv.Key != null && self.Owner.CanViewActor(kv.Value))
+				.Where(kv => kv.Key != null && (self.Owner.HasFogVisibility || self.Owner.CanViewActor(kv.Value)))
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: Use `Player.CanTargetActor` instead - that already includes this check and is more appropriate for use by `AutoTarget`.

Comment (penev92) [inline OpenRA.Mods.Common/Traits/AutoTarget.cs]: Good catch.

**Comment by penev92** [comment]
Changed.

**Comment by obrakmann** [comment]
:+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/d4f64e73e235bfc07ba4adb39047e43509c639b6)

**Pull request merged**